.github/workflows/ci_ec2_reusable.yml

```diff
@@ -95,51 +95,63 @@ jobs:
           ec2-instance-type: ${{ inputs.ec2_instance_type }}
           subnet-id: subnet-07b2729e5e065962f
           security-group-id: sg-0ab2e297196c8c381
-  functest:
-    name: Functional tests ${{ inputs.name }}
+  tests:
+    name: ${{ matrix.tests.name }} ${{ inputs.name }}
     needs: start-ec2-runner
     runs-on: ${{ needs.start-ec2-runner.outputs.label }}
-    if: ${{ inputs.functest }}
-    steps:
-      - uses: actions/checkout@v4
-      - uses: ./.github/actions/functest
-        with:
+    strategy:
+      fail-fast: false
+      matrix:
+        tests:
+        - name: Functional tests
+          action: functest
           nix-shell: ci
-          cflags: ${{ inputs.cflags }}
-          cross-prefix: ${{ inputs.cross-prefix }}
-          opt: ${{ inputs.opt }}
-  lint:
-    name: Linting ${{ inputs.name }}
-    needs: start-ec2-runner
-    runs-on: ${{ needs.start-ec2-runner.outputs.label }}
-    if: ${{ inputs.lint }}
-    steps:
-      - uses: actions/checkout@v4
-      - uses: ./.github/actions/lint
-        with:
+          run: ${{ inputs.functest }}
+        - name: Linting
+          action: lint
           nix-shell: ci-linter
-          cross-prefix: ${{ inputs.cross-prefix }}
-  cbmc:
-    name: CBMC ${{ inputs.name }}
-    needs: start-ec2-runner
-    runs-on: ${{ needs.start-ec2-runner.outputs.label }}
-    if: ${{ inputs.cbmc }}
+          run: ${{ inputs.lint }}
+        - name: CBMC
+          action: cbmc
+          nix-shell: ci-cbmc
+          run: ${{ inputs.cbmc }}
+        exclude:
+          - {tests: {
+              name: Functional tests,
+              action: functest,
+              nix-shell: ci,
+              run: false,
+            }}
+          - {tests: {
+              name: Linting,
+              action: lint,
+              nix-shell: ci-linter,
+              run: false,
+            }}
+          - {tests: {
+              name: CBMC,
+              action: cbmc,
+              nix-shell: ci-cbmc,
+              run: false,
+            }}
     steps:
       - uses: actions/checkout@v4
-      - uses: ./.github/actions/cbmc
+      - uses: jenseng/dynamic-uses@v1
```

Comment:
Why is this needed?

Reply:
as I pointed out in #155, currently with the `if` statement, github will still show the skipped jobs of a workflow in UI.

While with the `matrix` `exclude`, we can hide the skipped jobs from UI, but since the jobs we are running (functest/lint/cbmc) use different action, we'd need to dynamic load them.

And currently github doesn't support expression evaluation in `uses`, so the only way would be to write an action.yml on the fly as a middleware to dynamic pointing to use the specify action, which is what `jenseng/dynamic-uses` have done.

Reply:
Ok!

Reply:
btw, I haven't yet confirmed, but I think that when using matrix, if the whole matrix is excluded, there'd still be a dummy job executing (where all the matrix fields are evaluated to empty string).

In this case, I think we'd at least run one of functest/lint/cbmc job, so it'd be fine, but it's not the case for ec2 related job (which we'd like to exclude the jobs if triggered from fork). 

Therefore I didn't apply this to ec2 related jobs